src/actions/requests.rs

```diff
@@ -647,27 +649,18 @@ impl ExecuteCommand {
 /// of text by the server.
 pub struct CodeAction;
 
-impl<'a> Action<'a> for CodeAction {
-    type Params = CodeActionParams;
-    const METHOD: &'static str = "textDocument/codeAction";
-
-    fn new(_: &'a mut LsState) -> Self {
-        CodeAction
-    }
-}
-
-impl<'a> RequestAction<'a> for CodeAction {
-    type Response = Vec<Command>;
-    fn handle<O: Output>(&mut self, _id: usize, params: Self::Params, ctx_: &mut ActionContext, _out: O) -> Result<Self::Response, ()> {
-        trace!("code_action {:?}", params);
-
-        let ctx = ctx_.inited();
-        let file_path = parse_file_path!(&params.text_document.uri, "code_action")?;
-
-        let mut cmds = vec![];
-
+impl CodeAction {
+    /// Create CodeActions for fixes suggested by the compiler
+    /// the results are appended to `code_actions_result`
```

Comment:
I think it would be a little cleaner to return the actual value from function and `flat_map`/`collect` in CodeAction::handle, but that's just a matter of preference